submissions/ArthurGorbenko/Friend's app/js/script.js

```diff
@@ -0,0 +1,282 @@
+    buttons = [...buttons];
+    buttons.forEach(button => button.classList.add("button-dark"));
+  }
+  CONSTS.LIST_OF_CARDS.addEventListener("click", writeLetter);
+};
+
+const searchPeople = subString => {
+  if (!subString || subString === " ") {
+    return null;
+  }
+  if (!filteredByGender) {
+    users = usersConst;
+  }
+  subString = subString.toLowerCase();
+  let usersFound = users.filter(
+    user => user.name.first.match(subString) || user.name.last.match(subString)
+  );
+  createNewList(usersFound, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+};
+
+let query = "";
+
+const showGender = genderType => {
+  if (filteredByGender) {
+    users = usersConst;
+  }
+  users = users.filter(user => user.gender === genderType);
+  createNewList(users, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+  filteredByGender = true;
+  if (query) {
+    searchPeople(query);
+  }
+};
+
+const genderFilter = ({ target }) => {
+  if (target.checked) {
+    showGender(target.dataset.gender);
+  }
+};
+
+const sortingProcess = typeOfSort => {
+  const runSort = (a, b) => a.dob.age - b.dob.age;
+  if (typeOfSort === SORT_BY_ASCENDING) {
+    users.sort(runSort);
+  } else {
+    users.sort((a, b) => runSort(b, a));
+  }
+  createNewList(users, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+  searchPeople(query);
+};
+
+const ageSort = ({ target }) => {
+  if (target.checked) {
+    let sortType = target.dataset.render;
+    sortingProcess(sortType);
+  }
+};
+
+const resetRadio = allButtons => {
+  let buttons = [...allButtons];
+  buttons.forEach(button => (button.checked = false));
+};
+
+const resetAllOptions = () => {
+  CONSTS.SEARCH_FIELD.value = "";
+  query = "";
+  resetRadio(radioButtons);
+  createNewList(usersConst, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+};
+
+const switchTheme = ({ target }) => {
+  if (target.checked) {
+    makeDarkBackground();
+    changeColorCards("card", "button");
+  }
+};
+
+const addHamburgerOnMobile = () => {
+  let isMobile = window.matchMedia(MOBILE_RESOLUTION);
+  console.log(CONSTS.HAMBURGER);
+  if (!isMobile.matches) {
+    CONSTS.HAMBURGER.classList.add("visually-hidden");
+  } else {
+    return;
+  }
+};
+
+document.addEventListener("DOMContentLoaded", addHamburgerOnMobile);
+
+const start = () => {
+  CONSTS.HAMBURGER.addEventListener("click", moveContent);
+  CONSTS.SEND_BUTTON.addEventListener("click", messageSend);
+  CONSTS.RADIO_GENDER_FORM.addEventListener("change", genderFilter);
+  CONSTS.RADIO_AGE_FORM.addEventListener("change", ageSort);
+  CONSTS.RESET_BUTTON.addEventListener("click", resetAllOptions);
+  CONSTS.THEME_WRAPPER.addEventListener("change", switchTheme);
+  CONSTS.SEARCH_FIELD.addEventListener("keyup", function() {
+    query = this.value.toLowerCase();
+    searchPeople(query);
+  });
+};
+
+const getData = peoples => {
+  usersConst = peoples;
+  users = usersConst;
+  createNewList(users, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+  start();
+};
+
+fetch(
```

Comment:
It's preferable to make getData function with fetch inside. I will show you snippet from Ivan Tytarenko: 
`const getData = url =>
  fetch(url).then(res => {
    if (res.ok) {
      return res.json();
    }
    throw new Error(res.statusText);
  })
`

and using: 
`getData(url).then(peoples => {
    //in this part you can run code to handle people data
  });`

Reply:
Done.